dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramInstanceService.java

```diff
@@ -387,23 +388,9 @@ public ProgramInstance prepareProgramInstance( TrackedEntityInstance trackedEnti
         programInstance.setOrganisationUnit( organisationUnit );
         programInstance.enrollTrackedEntityInstance( trackedEntityInstance, program );
 
-        if ( enrollmentDate != null )
```

Comment:
I would not change code that handles null already just for the sake of transitioning to Optional.